### ISSUE TITLE
Implements wake on bluetooth for Steam Deck LCD

### ISSUE THREAD

**Issue by RodoMa92** [ublue-os/bazzite#2034]
It seems that Valve has implemented this feature also on Steam Deck LCD, but this requires a couple of changes into Bazzite to integrate it back:

1) This kernel patch [here](https://gitlab.com/evlaV/linux-integration/-/commit/dac16fb1d7192c9c23239c2881fc29283a73e697), not sure if it will ever reach upstream, but from the code alone this should not cause any issues with other hardware (there is both a hardware name check and a firmware revision check inside).
2) The upstream firmware need to be replaced with this one [here](https://gitlab.com/evlaV/linux-firmware-neptune/-/commit/9e7be9e0027f1f636833c225d83248da3ae24c51).
3) This new udev rule need to be added in, link [here](https://gitlab.com/evlaV/jupiter-hw-support/-/commit/7c460bc70fd3251576cd8ca8e509d323d61d8c84)

This should allow Bazzite to support wake on bluetooth on LCDs Decks.

**Comment by KyleGospo** [comment]
@antheas looping in for kernel patch

**Comment by antheas** [comment]
https://github.com/hhd-dev/kernel-bazzite/releases/tag/6.12.7-201

**Comment by RodoMa92** [comment]
Apparently they will push it upstream: https://gitlab.com/evlaV/linux-integration/-/commit/f22b0a48fcbbd02605de7427176d34addbb1c399

**Comment by patryk-matis** [comment]
> It seems that Valve has implemented this feature also on Steam Deck LCD, but this requires a couple of changes into Bazzite to integrate it back:
> 
> 1. This kernel patch [here](https://gitlab.com/evlaV/linux-integration/-/commit/dac16fb1d7192c9c23239c2881fc29283a73e697), not sure if it will ever reach upstream, but from the code alone this should not cause any issues with other hardware (there is both a hardware name check and a firmware revision check inside).
> 2. The upstream firmware need to be replaced with this one [here](https://gitlab.com/evlaV/linux-firmware-neptune/-/commit/9e7be9e0027f1f636833c225d83248da3ae24c51).
> 3. This new udev rule need to be added in, link [here](https://gitlab.com/evlaV/jupiter-hw-support/-/commit/7c460bc70fd3251576cd8ca8e509d323d61d8c84)
> 
> This should allow Bazzite to support wake on bluetooth on LCDs Decks.

No, Valve did say that this feature is only for the OLED version.
The issue is the bluetooth module.

**Comment by RodoMa92** [comment]
> > It seems that Valve has implemented this feature also on Steam Deck LCD, but this requires a couple of changes into Bazzite to integrate it back:
> > 
> > 1. This kernel patch [here](https://gitlab.com/evlaV/linux-integration/-/commit/dac16fb1d7192c9c23239c2881fc29283a73e697), not sure if it will ever reach upstream, but from the code alone this should not cause any issues with other hardware (there is both a hardware name check and a firmware revision check inside).
> > 2. The upstream firmware need to be replaced with this one [here](https://gitlab.com/evlaV/linux-firmware-neptune/-/commit/9e7be9e0027f1f636833c225d83248da3ae24c51).
> > 3. This new udev rule need to be added in, link [here](https://gitlab.com/evlaV/jupiter-hw-support/-/commit/7c460bc70fd3251576cd8ca8e509d323d61d8c84)
> > 
> > This should allow Bazzite to support wake on bluetooth on LCDs Decks.
> 
> No, Valve did say that this feature is only for the OLED version. The issue is the bluetooth module.

The issue is the firmware of the Bluetooth module, which has been already implemented by Valve themselves. The fact that they previously claimed that only the OLED would support this is irrelevant, since they did implement in the commit above.

More changes are needed, see this https://gitlab.com/evlaV/linux-integration/-/commit/3ee04c50b9b078289b990fb7d0680d88b69a25c4 and this https://gitlab.com/evlaV/linux-integration/-/commit/2317420b49c881ff99ce35e4d766457ae75a265e

**Comment by RodoMa92** [comment]
Tagging @antheas again to let it know to add these latest changes to the bazzite kernel

**Comment by noelmiller** [comment]
@antheas @KyleGospo any updates on this?

**Comment by KyleGospo** [comment]
> @antheas @KyleGospo any updates on this?

Likely just needs newer firmware, will revisit this soon.

**Comment by antheas** [comment]
I'm pretty sure we carry the patches. So if it doesn't work it's firmware and udev

**Comment by RodoMa92** [comment]
New fixed firmware dropped in the evlav repos: https://gitlab.com/evlaV/linux-firmware-neptune/-/commit/98edee542b81d2fdd09e4cefe0282223e069a8f3

**Comment by holopengin** [comment]
It seems like the only thing that's missing right now is the udev rule. If I manually add that file, I'm able to use Bluetooth wake via both my Dualshock 4 and Dualshock 4 V2.

I recall trying the same thing a while back but the steam deck would immediately wake up when I tried to sleep it. It seems that particular bug has been fixed. It now stays asleep until I turn the controller on.

**Comment by creedda** [comment]
Could we maybe add a ujust to add that udev rule? (Script could ask for input of which device is the Bluetooth device)

**Comment by jackedproxy** [comment]
Any update on this?
